### PR TITLE
Implement create_pretrained_project() for Pytorch engine

### PR DESCRIPTION
**Pull request by maximpavliv**
**This Pull Request adds Pytorch support for the `create_pretrained_project()` method (and closes https://github.com/DeepLabCut/DeepLabCut/issues/2732)**

Specifically:
- `create_pretrained_project()` becomes a `compat.py`-like method, and calls the specific method implementation based on the specified `engine` parameter. 
The old `create_pretrained_project()` is renamed to `create_pretrained_project_tensorflow()` and a new `create_pretrained_project_pytorch()` method with the Pytorch-shuffle logic is added.
The existing usages of `create_pretrained_project()` are also adapted.
- Both `create_pretrained_project_*` implementations call `_create_training_datasets_metadata()`. 
This method has been cherry-picked from #2881(where a first version, `_create_and_save_fake_metadata()`, has been implemented as a temporary fix), and adapted to work with both Tensorflow and Pytorch. It creates the `metadata.yaml ` file in the training-datasets directory, and fills it with a first shuffle corresponding to the downloaded modelzoo model.
- The video processing logic (that is optionally done after the pretrained project has been created) is extracted to 
`_process_videos()`, and called by both `create_pretrained_project_*` implementations.
- A new `individuals` parameter is added to `create_new_project()` and to `create_pretrained_project()`, so that the list of individuals can be automatically filled in the different configuration files of the new project.
- An integration test script for `create_pretrained_project(engine=Engine.PYTORCH)` is added.

**This Pull Request also does some fixes to issues that have been spotted during its development:**
- Fixes `list_videos_in_folder()` : in original (Tensorflow-implementation) methods (f.ex. `analyze_videos()`), `videotype` argument's value by default was `""` (instead of `None`), and this default argument value was not treated properly.
- `get_super_animal_project_cfg()` was using `read_config()` to read the superanimal project configuration template embedded into the DeepLabCut repository, which then modified the file in the DeepLabCut codebase. Now, the method reads the template with `read_config_as_dict()`, which doesn't modify the file.
- The previous fix created a circular-import issue, that has been resolved by moving the `read_config_as_dict()`, `write_config()` and `pretty_print()` methods to `deeplabcut.core.config_utils.py` module.
- The other superanimal integration test scripts (`testscript_superanimal_adaptation.py`, `testscript_superanimal_inference.py` and `testscript_superanimal_transfer_learning.py`) have been fixed.

### CODE REVIEW

**Comment by maximpavliv** [comment]
The Tensorflow implementation of `create_pretrained_project()` fills the `"default_net_type"` field in the project configuration file with the chosen modelzoo's model architecture. 
In the Pytorch implementation of `create_pretrained_project()` I wasn't sure how to do this, because we would also then need a "default detector type" field (particularly since Superanimal models are all Top-Down currently), but such a field doesn't exist. Therefore I left the `"default_net_type"` field  emtpy for now. 
Any suggestions welcome on this point.

File: deeplabcut/create_project/new.py [hunk 1]
```diff
@@ -58,6 +59,9 @@ def create_new_project(
     multianimal: bool, optional. Default: False.
         For creating a multi-animal project (introduced in DLC 2.2)
 
+    individuals: list[str]|None = None,
+        todo
+
```
Comment (n-poulsen) [inline deeplabcut/create_project/new.py]: We could add a docstring here

**Review by n-poulsen — APPROVED** [review]
LGTM @maximpavliv, thanks! just one minor comment

File: deeplabcut/create_project/new.py (hunk 1 above)
Comment (maximpavliv) [reply deeplabcut/create_project/new.py]: Right, I forgot!
